internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/fastclient/FastClientServerReadQuotaTest.java

```diff
@@ -39,22 +40,30 @@ public void testServerReadQuota() throws Exception {
         assertEquals((int) value.get(VALUE_FIELD_NAME), i);
       }
     }
-    MetricsRepository serverMetric = veniceCluster.getVeniceServers().get(0).getMetricsRepository();
+    ArrayList<MetricsRepository> serverMetrics = new ArrayList<>();
+    for (int i = 0; i < veniceCluster.getVeniceServers().size(); i++) {
+      serverMetrics.add(veniceCluster.getVeniceServers().get(i).getMetricsRepository());
+    }
     String readQuotaRequestedString = "." + storeName + "--quota_rcu_requested.Count";
     String readQuotaRejectedString = "." + storeName + "--quota_rcu_rejected.Count";
     String readQuotaUsageRatio = "." + storeName + "--read_quota_usage_ratio.Gauge";
     TestUtils.waitForNonDeterministicAssertion(10, TimeUnit.SECONDS, () -> {
-      assertNotNull(serverMetric.getMetric(readQuotaRequestedString));
-      assertNotNull(serverMetric.getMetric(readQuotaRejectedString));
-      assertNotNull(serverMetric.getMetric(readQuotaUsageRatio));
+      for (MetricsRepository serverMetric: serverMetrics) {
+        assertNotNull(serverMetric.getMetric(readQuotaRequestedString));
+        assertNotNull(serverMetric.getMetric(readQuotaRejectedString));
+        assertNotNull(serverMetric.getMetric(readQuotaUsageRatio));
+      }
     });
-    assertTrue(
-        serverMetric.getMetric(readQuotaRequestedString).value() >= 250,
-        "Metric value: " + serverMetric.getMetric(readQuotaRequestedString).value());
-    assertEquals(serverMetric.getMetric(readQuotaRejectedString).value(), 0d);
-    assertTrue(serverMetric.getMetric(readQuotaUsageRatio).value() > 0);
+    int averageQuotaRequested = 500 / serverMetrics.size();
+    for (MetricsRepository serverMetric: serverMetrics) {
+      assertTrue(
+          serverMetric.getMetric(readQuotaRequestedString).value() >= averageQuotaRequested,
```

Comment:
This assumes the traffic is distributed uniformly, right? Is the assumption too tight? Should we check that the sum of that metric across all servers is >= 500 instead?

Reply:
It's pretty uniform based on the current routing strategy but sure if it's just for the purpose of verifying quota we can do the sum instead.